test/gnmi/offlineintopotest.go

```diff
@@ -0,0 +1,110 @@
+	"github.com/onosproject/onos-test/pkg/onit/env"
+	"github.com/onosproject/onos-topo/api/device"
+	"github.com/openconfig/gnmi/proto/gnmi_ext"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	offlineInTopoModPath          = "/system/clock/config/timezone-name"
+	offlineInTopoModValue         = "Europe/Rome"
+	offlineInTopoModDeviceName    = "offline-dev-1"
+	offlineInTopoModDeviceVersion = "1.0.0"
+	offlineInTopoModDeviceType    = "Devicesim"
+)
+
+// TestOfflineDeviceInTopo tests set/query of a single GNMI path to a single device that is in the config but offline
+func (s *TestSuite) TestOfflineDeviceInTopo(t *testing.T) {
+
+	deviceClient, deviceClientError := env.Topo().NewDeviceServiceClient()
+	assert.NotNil(t, deviceClient)
+	assert.Nil(t, deviceClientError)
+	newDevice := &device.Device{
+		ID:          offlineInTopoModDeviceName,
+		Revision:    0,
+		Address:     offlineInTopoModDeviceName + ":10161",
+		Target:      "",
+		Version:     offlineInTopoModDeviceVersion,
+		Timeout:     nil,
+		Credentials: device.Credentials{},
+		TLS:         device.TlsConfig{},
+		Type:        offlineInTopoModDeviceType,
+		Role:        "",
+		Attributes:  nil,
+		Protocols:   nil,
+	}
+	addRequest := &device.AddRequest{Device: newDevice}
+	addResponse, addResponseError := deviceClient.Add(context.Background(), addRequest)
+	assert.NotNil(t, addResponse)
+	assert.Nil(t, addResponseError)
+
+	simulator := env.NewSimulator().SetName(offlineInTopoModDeviceName)
+
+	// Make a GNMI client to use for requests
+	c, err := env.Config().NewGNMIClient()
+	assert.NoError(t, err)
+	assert.True(t, c != nil, "Fetching client returned nil")
+
+	// Set a value using gNMI client to the offline device
+	extNameDeviceType := gnmi_ext.Extension_RegisteredExt{
+		RegisteredExt: &gnmi_ext.RegisteredExtension{
+			Id:  gnmi.GnmiExtensionDeviceType,
+			Msg: []byte("Devicesim"),
+		},
+	}
+	extNameDeviceVersion := gnmi_ext.Extension_RegisteredExt{
+		RegisteredExt: &gnmi_ext.RegisteredExtension{
+			Id:  gnmi.GnmiExtensionVersion,
+			Msg: []byte(offlineInTopoModDeviceVersion),
+		},
+	}
+	extensions := []*gnmi_ext.Extension{{Ext: &extNameDeviceType}, {Ext: &extNameDeviceVersion}}
+
+	setPath := makeDevicePath(offlineInTopoModDeviceName, offlineInTopoModPath)
+	setPath[0].pathDataValue = offlineInTopoModValue
+	setPath[0].pathDataType = StringVal
+
+	_, extensionsSet, errorSet := gNMISet(MakeContext(), c, setPath, noPaths, extensions)
+	assert.NoError(t, errorSet)
+	assert.Equal(t, 1, len(extensionsSet))
+	extensionBefore := extensionsSet[0].GetRegisteredExt()
+	assert.Equal(t, extensionBefore.Id.String(), strconv.Itoa(gnmi.GnmiExtensionNetwkChangeID))
+
+	//  Start the simulator
+	simulatorEnv := simulator.AddOrDie()
+	time.Sleep(2 * time.Second)
+
+	// Check that the value was set correctly
+	valueAfter, extensions, errorAfter := gNMIGet(MakeContext(), c, makeDevicePath(offlineInTopoModDeviceName, offlineInTopoModPath))
```

Comment:
I think this `Get` could be done before the Simulator is started above

The get will read from onos-config only and not from the device.
It would also be good to check the state of the `NetworkChange` at this stage - it should be in `PENDING`